webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/smartling/SmartlingResultProcessor.java

```diff
@@ -0,0 +1,105 @@
+package com.box.l10n.mojito.service.thirdparty.smartling;
+
+import com.amazonaws.SdkClientException;
+import com.box.l10n.mojito.service.blobstorage.Retention;
+import com.box.l10n.mojito.service.blobstorage.s3.S3BlobStorage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+@ConditionalOnProperty(value = "l10n.ThirdPartyTMS.impl", havingValue = "ThirdPartyTMSSmartling")
+@Component
+public class SmartlingResultProcessor {
+
+    static Logger logger = LoggerFactory.getLogger(SmartlingResultProcessor.class);
+
+    private static final String NAME_FORMAT = "%s_%s_%s.zip";
+
+    @Autowired
+    S3BlobStorage s3BlobStorage;
+
+    public String processPush(List<SmartlingFile> files,
+                              SmartlingOptions options) {
+        return processAction(files, options, "push");
+    }
+
+    public String processPushTranslations(List<SmartlingFile> files,
+                                          SmartlingOptions options){
+        return processAction(files, options, "push_translations");
+    }
+
+    private String processAction(List<SmartlingFile> files,
+                                 SmartlingOptions options,
+                                 String action) {
+        String result = null;
+
+        if (!files.isEmpty() && options.isDryRun() && s3BlobStorage != null){
+            try {
+                result = uploadFiles(files, options.getRequestId(), action);
+                logger.info("{} result for request id {} uploaded to: {}", action, options.getRequestId(), result);
```

Comment:
debug?

Reply:
Doesn't seems like a good option. We'll need to turn on/off debug logging just for this purpose, and this one line is required to get the s3 URL to be downloaded.

Reply:
we're trying (might not be always consistent) to use debug for repeatable messages and info should be more for one off, like framework initialization. 

Not sure to understand how that system would work? tailing the log would require to have access to the host. it is impractical, no?

Can the URL be guess or displayed in the CLI output?

Reply:
We can obtain this log event by tailing the log file and grepping the `requestId` string, or through Kibana when enabled. And we can guess the URL but eventually, that URL shouldn't be guessed and rather logged so we can get it easily when running tests with a dry run.

Reply:
Most likely there is some kind of lag in the log propagation and having a predictable URL seem like better option to me. Up to you, since this is mostly for testing